.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,31 @@
+{
```

Comment:
Can you investigate this flow:

1. Use RBAC
2. Store and Use Keys in KeyVault.  If the KV isn't specified, then try to read the values from the service with listKeys.

The above adds more complexity, but allows the user to store the keys in KV as an option.

Reply:
Just linking this as we should revisit the Speech Service using this method too: - https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/issues/101 (not for this PR!)

Reply:
Use of `KeyVault` has been added and is currently testing well.

Reply:
Sorry for the back and forth on this.  Let's do this:

Default to using rbac.

If AUTH_TYPE is rbac, then assume the developer wants to also use key vault.  Put the keys in keyvault and then read them from keyvault.

If the user sets AUTH_TYPE to keys, then read the keys from the environment and set the keys using listKeys function.

I think that will be cleaner.

Reply:
In our understanding, if `AUTH_TYPE` is `rbac`, then some `App Services` and `User` should be assigned corresponding permissions in the `main.bicep` file. This means that keys will no longer be needed in the code, but the service will be accessed directly through something like `DefaultAzureCredential()`.

So regarding what you mention here, `If AUTH_TYPE is rbac, then assume the developer wants to also use key vault.`, which confuses us. Because in our concept, one of them is enough.

If we have any misunderstandings about `rbac` or your comments, please let me know, thanks.

Reply:
Yes, only use a key vault if there's no RBAC option for the service.